torch_np/_dtypes.py

```diff
@@ -0,0 +1,174 @@
+""" Define the supported dtypes and numpy <--> torch dtype mapping, define casting rules. 
+"""
+
+# TODO: 1. define torch_np dtypes, make this work without numpy.
+#       2. mimic numpy's various aliases (np.half == np.float16, dtype='i8' etc)
+#       3. convert from python types: np.ones(3, dtype=float) etc
+
+import builtins
+import torch
+
+# Define analogs of numpy dtypes supported by pytorch.
+
+class dtype:
```

Comment:
the `dtype` class is a super-minimal duck-typing analog of numpy dtypes. Instances are not singletons! `dtype("int64") == dtype("int64")` but these are two different instances.